src/commands/PublishCommand.js

```diff
@@ -342,6 +352,13 @@ export default class PublishCommand extends Command {
       // write new package
       FileSystemUtilities.writeFileSync(packageJsonLocation, pkg.toJsonString());
 
+      // we can now generate the Changelog, based on the
+      // the updated version that we're about to release.
+      if (this.flags.conventionalCommits) {
+        ConventionalCommitUtilties.updateChangelog(update.package.name, changelogLocation);
+        changedFiles.push(changelogLocation);
```

Comment:
I would prefer to encapsulate the `changelogLocation` assignment inside `updateChangelog()`. Perhaps return the changed location for pushing into `changedFiles`, then?

In fact, changelogs being important enough, regardless of how one generates them, I can justify adding a `changelogLocation` getter to the [Package](https://github.com/lerna/lerna/blob/master/src/Package.js) class, thus removing the need for a return value.

```diff
diff --git a/src/Package.js b/src/Package.js
index 76a88da..d4d4bee 100644
--- a/src/Package.js
+++ b/src/Package.js
@@ -17,6 +17,10 @@ export default class Package {
     return this._location;
   }
 
+  get changelogLocation() {
+    return path.join(this._location, "CHANGELOG.md");
+  }
+
   get nodeModulesLocation() {
     return path.join(this._location, "node_modules");
   }
```